daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilXMLLoader.scala

```diff
@@ -538,47 +401,226 @@ trait SchemaAwareLoaderMixin {
+    // f.setFeature("http://xml.org/sax/features/validation/dynamic", true)
+    f.setFeature("http://apache.org/xml/features/honour-all-schemaLocations", true)
+    f.setFeature("http://apache.org/xml/features/validation/schema", true)
+    f.setFeature("http://apache.org/xml/features/validation/schema-full-checking", true)
+    f
+  }
+
+  private lazy val noSchemaParser: SAXParser = {
+    parserFromFactory(parserFactory())
+  }
+
+  private def parserFromFactory(f: SAXParserFactory) = {
+    val p = f.newSAXParser()
+    //
+    // Not allowed on a SAXParser
+    // p.setProperty(XMLUtils.SAX_NAvMESPACES_FEATURE, true)
+    // Not allowed on a SAXParser
+    // p.setProperty(XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE, true)
+    val xrdr = p.getXMLReader()
+    XMLUtils.setSecureDefaults(xrdr)
+    xrdr.setErrorHandler(errorHandler)
+    // not recognized by XMLReader
+    // xrdr.setFeature("http://xml.org/sax/features/validation/dynamic", true)
+    xrdr.setContentHandler(this)
+    //
+    // This is required to get the parse to really use our resolver.
+    // The setEntityResolver(resolver) does not work.
+    //
+    xrdr.setProperty("http://apache.org/xml/properties/internal/entity-resolver", resolver)
+    p
+  }
+
+  /**
+   * This is the common routine called by all the load calls to actually
+   * carry out the loading of the schema.
+   *
+   * There are no calls to this in Daffodil code base as of this writing, but
+   * the base class scala.xml.factory.XMLLoader calls it.
+   *
    * Does (optional) validation,
+   *
+   * @param source The URI for the XML document which may be a XML or DFDL schema, or just XML data.
+   * @param optSchemaURI Optional URI for XML schema for the XML source document.
+   * @param addPositionAttributes True to add dafint:file dafint:line attributes to all elements.
+   *                              Defaults to false.
+   * @param normalizeCRLFtoLF True to normalize CRLF and isolated CR to LF. This should usually be true,
+   *                          but some special case situations may require preservation of CRLF/CR.
+   * @return an scala.xml.Node (Element actually) which is the document element of the source.
    */
-  def load(source: DaffodilSchemaSource): scala.xml.Node = {
-    if (doValidation) {
-      val inputSource = source.newInputSource()
-      val xercesNode = xercesAdapter.load(inputSource) // validates
-      inputSource.getByteStream().close()
-
-      if (xercesNode == null) return null
-      // Note: we don't call xercesAdapter.validateSchema(source)
-      // here, because this is an XML loader, not necessarily
-      // just a DFDL schema loader. So for example the doValidation flag
-      // above could be telling us to validate a TDML file or not.
+  def load(source: DaffodilSchemaSource,
+    optSchemaURI: Option[URI],
+    addPositionAttributes: Boolean = false,
+    normalizeCRLFtoLF: Boolean = true): scala.xml.Node = {
+    //
```

Comment:
Note @jw3 we're using the XercesValidator that is built in, here so that we have only one instance of Xerces validation now.